alpha/apps/kaltura/lib/kUploadTokenMgr.php

```diff
@@ -168,14 +170,56 @@ public function uploadFileToToken($fileData, $resume = false, $resumeAt = -1)
 		}
 		else
 		{
-			//We return null file size when we want to faile the upload since it reached max chunks waiting for concat
+			//We return null file size when we want to fail the upload since it reached max chunks waiting for concat
 			$this->_uploadToken->setStatus(!is_null($fileSize) ? UploadToken::UPLOAD_TOKEN_PARTIAL_UPLOAD : UploadToken::UPLOAD_TOKEN_ERROR);
 		}
 		
 		$this->_uploadToken->setUploadedFileSize($fileSize);
 		$this->_uploadToken->setDc(kDataCenterMgr::getCurrentDcId());
 		
 		$this->_uploadToken->save();
+		
+		if ($this->_uploadToken->getObjectId() && $this->_uploadToken->getObjectType())
+		{
+			switch ($this->_uploadToken->getObjectType())
+			{
+				case self::ENTRY:
+				{
+					$entry = entryPeer::retrieveByPK($this->_uploadToken->getObjectId());
+					break;
+				}
+				case self::FLAVOR:
+				{
+					$flavorAsset = assetPeer::retrieveById($this->_uploadToken->getObjectId());
+					if ($flavorAsset && $flavorAsset->getType() == assetType::FLAVOR && $flavorAsset->getEntryId())
+					{
+						$entry = entryPeer::retrieveByPK($flavorAsset->getEntryId());
+					}
+					break;
+				}
+			}
+			if ($entry && $entry->getType() == entryType::MEDIA_CLIP && !$entry->getMediaType())
```

Comment:
$entry is not necesseraly defined, declere it before the call to the new function you will create